### ISSUE TITLE
Database: Enabled database metrics feature flag corrupts MySQL database tables on restart

### ISSUE THREAD

**Issue by papagian** [grafana/grafana#28440]
**What happened**:
Enabled the feature flag `database_metrics` and restarted Grafana. Then restarted Grafana again and the MySQL database migrations failed and .

**What you expected to happen**:
Grafana to start and MySQL database migrations to run successfully and don't corrupt any database tables.

**How to reproduce it (as minimally and precisely as possible)**:
1. start mysql:
`make devenv sources=mysql`
2. run grafana (7.2.2) using:
`GF_LOG_LEVEL=debug  GF_DATABASE_TYPE=mysql GF_DATABASE_HOST=127.0.0.1:3306 GF_DATABASE_USER=grafana GF_DATABASE_PASSWORD=<password> GF_FEATURE_TOGGLES_ENABLE=database_metrics GF_LOG_FILTERS=DBMetrics:debug make run`

2. wait for the server to start
3. restart the server
4. the server fails to start because tries to rerun migrations and reports database errors

**Anything else we need to know?**:
I have tried also running with postgreSQL and sqlite and I'm unable to reproduce it.
Also I have tried the above scenario without the feature flag enabled and I can't reproduce it.

**Environment**:
- Grafana version: 7.2.2 (and master before #28438 was merged)

**Comment by papagian** [comment]
Digging a bit deeper into it; I suspect that the cause is the `"driver: skip fast-path; continue as if unimplemented"` error when restarting grafana with MySQL database and with `GF_FEATURE_TOGGLES_ENABLE=database_metrics`

the log extract when starting grafana with empty database with the feature toggle enabled:
```
t=2020-10-22T13:22:05+0300 lvl=info msg="Starting DB migrations" logger=migrator
t=2020-10-22T13:22:05+0300 lvl=dbug msg="query finished" logger=sqlstore.metrics status=success elapsed time=6.752µs sql="SELECT `TABLE_NAME` from `INFORMATION_SCHEMA`.`TABLES` WHERE `TABLE_SCHEMA`=? and `TABLE_NAME`=?" error="driver: skip fast-path; continue as if unimplemented"
t=2020-10-22T13:22:05+0300 lvl=dbug msg="query finished" logger=sqlstore.metrics status=success elapsed time=1.236763ms sql="SELECT `TABLE_NAME` from `INFORMATION_SCHEMA`.`TABLES` WHERE `TABLE_SCHEMA`=? and `TABLE_NAME`=?"
t=2020-10-22T13:22:05+0300 lvl=info msg="Executing migration" logger=migrator id="create migration_log table"
```

log extract when restarting grafana with the feature toggle enabled:
```
t=2020-10-22T13:22:29+0300 lvl=info msg="Starting DB migrations" logger=migrator
t=2020-10-22T13:22:29+0300 lvl=dbug msg="query finished" logger=sqlstore.metrics status=success elapsed time=5.877µs sql="SELECT `TABLE_NAME` from `INFORMATION_SCHEMA`.`TABLES` WHERE `TABLE_SCHEMA`=? and `TABLE_NAME`=?" error="driver: skip fast-path; continue as if unimplemented"
t=2020-10-22T13:22:29+0300 lvl=dbug msg="query finished" logger=sqlstore.metrics status=success elapsed time=1.77814ms sql="SELECT `TABLE_NAME` from `INFORMATION_SCHEMA`.`TABLES` WHERE `TABLE_SCHEMA`=? and `TABLE_NAME`=?"
t=2020-10-22T13:22:29+0300 lvl=info msg="Executing migration" logger=migrator id="create migration_log table"
```

whereas the log extract when starting grafana with empty database without the feature toggle enabled:
```
t=2020-10-22T13:51:45+0300 lvl=info msg="Starting DB migrations" logger=migrator
t=2020-10-22T13:51:45+0300 lvl=info msg="Executing migration" logger=migrator id="create migration_log table"
```

 and the log extract enabled when restarting grafana without the feature toggle :
```
t=2020-10-22T13:52:04+0300 lvl=info msg="Starting DB migrations" logger=migrator
t=2020-10-22T13:52:04+0300 lvl=dbug msg="Skipping migration: Already executed" logger=migrator id="create migration_log table"
```

**Comment by papagian** [comment]
These are the logs when I run grafana with PostgreSQL database and with `GF_FEATURE_TOGGLES_ENABLE=database_metrics`:

when starting grafana with empty database:
```
t=2020-10-22T13:23:58+0300 lvl=info msg="Starting DB migrations" logger=migrator
t=2020-10-22T13:23:58+0300 lvl=dbug msg="query finished" logger=sqlstore.metrics status=success elapsed time=3.144332ms sql="SELECT tablename FROM pg_tables WHERE tablename = $1"
t=2020-10-22T13:23:58+0300 lvl=info msg="Executing migration" logger=migrator id="create migration_log table"
```

when restarting grafana:
```
t=2020-10-22T13:24:22+0300 lvl=info msg="Starting DB migrations" logger=migrator
t=2020-10-22T13:24:22+0300 lvl=dbug msg="query finished" logger=sqlstore.metrics status=success elapsed time=3.712142ms sql="SELECT tablename FROM pg_tables WHERE tablename = $1"
t=2020-10-22T13:24:22+0300 lvl=dbug msg="query finished" logger=sqlstore.metrics status=success elapsed time=2.69571ms sql="SELECT \"id\", \"migration_id\", \"sql\", \"success\", \"error\", \"timestamp\" FROM \"migration_log\""
t=2020-10-22T13:24:22+0300 lvl=dbug msg="Skipping migration: Already executed" logger=migrator id="create migration_log table"
```

**Comment by papagian** [comment]
Fixed by #29662
